nomos-libp2p/src/behaviour/nat/state_machine/transitions.rs

```diff
@@ -0,0 +1,165 @@
+use crate::behaviour::nat::state_machine::{
+    event::Event,
+    states::{
+        MappedPublic, Private, Public, TestIfMappedPublic, TestIfPublic, TryMapAddress,
+        Uninitialized,
+    },
+    Command, CommandTx, OnEvent, State,
+};
+
+impl OnEvent for State<Uninitialized> {
```

Comment:
Do you mind adding a comment on top of each state transition explaining what events they react to, under what conditions, and in which state they end up being?